exchanges/binanceus/binanceus.go

```diff
@@ -0,0 +1,1966 @@
+	}
+	if arg.Limit > 0 && arg.Limit < 1000 {
+		params.Set("limit", fmt.Sprint(arg.Limit))
+	} else if arg.Limit > 1000 {
+		params.Set("limit", strconv.Itoa(1000))
+	}
+	return resp, bi.SendAuthHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, myTrades, params, spotDefaultRate, &resp)
+}
+
+// OCO Orders
+
+// CreateNewOCOOrder o place a new OCO(one-cancels-the-other) order.
+func (bi *Binanceus) CreateNewOCOOrder(ctx context.Context, arg *OCOOrderInputParams) (*OCOFullOrderResponse, error) {
+	params := url.Values{}
+	if arg == nil || arg.Symbol == "" || len(arg.Symbol) <= 2 || arg.Quantity == 0 || arg.Side == "" || arg.Price == 0 || arg.StopPrice == 0 {
+		return nil, errIncompleteArguments
+	}
+	params.Set("symbol", arg.Symbol)
+	params.Set("quantity", strconv.FormatFloat(arg.Quantity, 'f', 5, 64))
+	params.Set("side", arg.Side)
+	params.Set("price", strconv.FormatFloat(arg.Price, 'f', 5, 64))
+	params.Set("stopPrice", strconv.FormatFloat(arg.StopPrice, 'f', 5, 64))
+	if arg.ListClientOrderID != "" {
+		params.Set("listClientOrderId", arg.ListClientOrderID)
+	}
+	if arg.LimitClientOrderID != "" {
+		params.Set("limitClientOrderId", arg.LimitClientOrderID)
+	}
+	if arg.LimitIcebergQty > 0 {
+		params.Set("limitIcebergQty", strconv.FormatFloat(arg.LimitIcebergQty, 'f', 5, 64))
+	}
+	if arg.StopClientOrderID != "" {
+		params.Set("stopClientOrderId", arg.StopClientOrderID)
+	}
+	if arg.StopLimitPrice > 0.0 {
+		params.Set("stopLimitPrice", strconv.FormatFloat(arg.StopLimitPrice, 'f', 5, 64))
+	}
+	if arg.StopIcebergQty > 0.0 {
+		params.Set("stopIcebergQty", strconv.FormatFloat(arg.StopIcebergQty, 'f', 5, 64))
+	}
+	if arg.StopLimitTimeInForce != "" {
+		params.Set("stopLimitTimeInForce", arg.StopLimitTimeInForce)
+	}
+	if arg.NewOrderRespType != "" {
+		params.Set("newOrderRespType", arg.NewOrderRespType)
+	}
+	if arg.RecvWindow > 200 {
+		params.Set("recvWindow", strconv.Itoa(int(arg.RecvWindow)))
+	} else {
+		params.Set("recvWindow", "6000")
+	}
+	params.Set("timestamp", strconv.FormatInt(time.Now().UnixMilli(), 10))
+	var response OCOFullOrderResponse
+	return &response, bi.SendAuthHTTPRequest(ctx,
+		exchange.RestSpotSupplementary,
+		http.MethodPost, ocoOrder, params,
+		spotDefaultRate, &response)
+}
+
+// GetOCOOrder to retrieve a specific OCO order based on provided optional parameters.
+func (bi *Binanceus) GetOCOOrder(ctx context.Context, arg *GetOCOOrderRequestParams) (*OCOOrderResponse, error) {
+	params := url.Values{}
+	params.Set("timestamp", fmt.Sprint(time.Now().UnixMilli()))
+	switch {
+	case arg.OrderListID != "":
+		params.Set("orderListId", arg.OrderListID)
+	case arg.OrigClientOrderID != "":
+		params.Set("origClientOrderId", arg.OrigClientOrderID)
+	default:
+		return nil, errIncompleteArguments
+	}
+	params.Set("recvWindow", "60000")
+	var response OCOOrderResponse
+	return &response, bi.SendAuthHTTPRequest(ctx, exchange.RestSpotSupplementary, http.MethodGet, ocoOrderList, params, spotDefaultRate, &response)
```

Comment:
The GetOCOOrder ratelimit for orders is 2. See https://docs.binance.us/#get-oco-order
I'm going to leave it to you to review the remaining endpoints rather than bring up all of them.